### PR TITLE
update for py-elephant

### PR DESCRIPTION
**Pull request by Moritz-Alexander-Kern**
add new versions to py-elephant

### CODE REVIEW

**Review by alalazo — CHANGES_REQUESTED** [review]
```
PKG-DIRECTIVES: 2 issues found
1. py-elephant: dependency on py-neo@0.10.0: cannot be satisfied by known versions of py-neo
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-elephant/package.py
    known versions of py-neo are 0.8.0, 0.5.2, 0.4.1, 0.3.3
2. py-elephant: dependency on py-neo@0.[9](https://github.com/spack/spack/runs/6385190604?check_suite_focus=true#step:6:9).0 cannot be satisfied by known versions of py-neo
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-elephant/package.py
    known versions of py-neo are 0.8.0, 0.5.2, 0.4.1, 0.3.3
```

File: var/spack/repos/builtin/packages/py-elephant/package.py [hunk 1]
```diff
@@ -10,20 +10,45 @@ class PyElephant(PythonPackage):
...
+    version('0.11.0', sha256='7b547964dbd196361edc922db2c5a7c0c886ef1effcca6c6dc7adb06f966a3be')
+    version('0.10.0', sha256='7e69a113475e4db01b3563328953c037d37f1597d9f2edf0d51fb65e9aebf096')
+    version('0.9.0', sha256='3e3d4a8e45d708f48bdcadcc4933c66f757d1ede6a1e172af0c07331b64ca180')
+    version('0.8.0', sha256='f7c2649d5b7cfdbaa4442457c75f86af01cc8e7ce2c63f5b3d4687bb94e10af2')
+    version('0.7.0', sha256='76785fe10c40042504928fde2fc57182230bbe39cf0fb0dcaffaba76219b046a')
+    version('0.6.4', sha256='b8c5f2c00ad3249e1fe428d0b8a1dbcaee4a69464481f5f8fd55d2f7f22c45a3')
     version('0.4.1', sha256='86b21a44cbacdc09a6ba6f51738dcd5b42ecd553d73acb29f71a0be7c82eac81')
     version('0.3.0', sha256='747251ccfb5820bdead6391411b5faf205b4ddf3ababaefe865f50b16540cfef')
 
     variant('doc', default=False, description='Build the documentation')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-elephant/package.py]:
```suggestion
    variant('docs', default=False, description='Install documentation dependencies')
```
This doesn't actually build the docs, it just makes sure the deps are installed. Also, let's call it `+docs` to match `pip install elephant[docs]`

**Review by alalazo — CHANGES_REQUESTED** [review]
```
PKG-DIRECTIVES: 3 issues found
1. py-elephant: wrong variant in "depends_on" directive
    the variant 'doc' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-elephant/package.py
2. py-elephant: dependency on py-neo@0.10.0: cannot be satisfied by known versions of py-neo
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-elephant/package.py
    known versions of py-neo are 0.8.0, 0.5.2, 0.4.1, 0.3.3
3. py-elephant: dependency on py-neo@0.[9](https://github.com/spack/spack/runs/6401724866?check_suite_focus=true#step:6:9).0 cannot be satisfied by known versions of py-neo
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/py-elephant/package.py
    known versions of py-neo are 0.8.0, 0.5.2, 0.4.1, 0.3.3
```

File: var/spack/repos/builtin/packages/py-elephant/package.py [hunk 2]
```diff
@@ -10,20 +10,45 @@ class PyElephant(PythonPackage):
...
+
+    depends_on('py-setuptools', type='build')
+    depends_on('python@3.7:', type=('build', 'run'), when='@0.11.0:')
+    depends_on('py-neo@0.3.4:', type=('build', 'run'), when='@0.3.0:0.4.1')  # > 0.3.3 ?
+    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-quantities@0.10.1:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-scipy@0.14.0:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-pandas@0.14.1:', type=('build', 'run'), when='+pandas')
+    depends_on('py-numpydoc@0.5:', type=('build', 'run'), when='+doc')
+    depends_on('py-sphinx@1.2.2:', type=('build', 'run'), when='+doc')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-elephant/package.py]:
```suggestion
    depends_on('py-numpydoc@0.5:', type=('build', 'run'), when='+docs')
    depends_on('py-sphinx@1.2.2:', type=('build', 'run'), when='+docs')
```

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-elephant/package.py [hunk 3]
```diff
@@ -3,27 +3,52 @@
...
+
+    depends_on('py-setuptools', type='build')
+    depends_on('python@3.7:', type=('build', 'run'), when='@0.11.0:')
+    depends_on('py-neo@0.3.4:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-quantities@0.10.1:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-scipy@0.14.0:', type=('build', 'run'), when='@0.3.0:0.4.1')
+    depends_on('py-pandas@0.14.1:', type=('build', 'run'), when='+pandas')
+    depends_on('py-numpydoc@0.5:', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx@1.2.2:', type=('build', 'run'), when='+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-elephant/package.py]: Looks like the docs requirements have changed in the latest release:
```
numpydoc>=1.1.0
jupyter>=1.0.0
sphinx>=3.3.0
nbsphinx>=0.8.0
sphinxcontrib-bibtex==1.0.0
sphinx-tabs>=1.3.0
matplotlib>=3.3.2
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-elephant/package.py (hunk 3 above)
Comment (Moritz-Alexander-Kern) [reply var/spack/repos/builtin/packages/py-elephant/package.py]: see: https://github.com/spack/spack/pull/30613/commits/b1cc029016af2a2095a425fb4cfa672f7783f2c4

File: var/spack/repos/builtin/packages/py-elephant/package.py [hunk 4]
```diff
@@ -3,27 +3,63 @@
...
+    depends_on('py-pandas@0.14.1:', type=('build', 'run'), when='+pandas')
+    depends_on('py-numpydoc@0.5:', type=('build', 'run'), when='@0.3.0:0.8.0+docs')
+    depends_on('py-numpydoc@1.1.0:', type=('build', 'run'), when='@0.9.0:+docs')
+    depends_on('py-sphinx@1.2.2:', type=('build', 'run'), when='@0.3.0:0.6.0+docs')
+    depends_on('py-sphinx@2.4.3:', type=('build', 'run'), when='@0.7.0:0.8.0+docs')
+    depends_on('py-sphinx@3.3.0:', type=('build', 'run'), when='@0.9.0:+docs')
+    depends_on('py-jupyter@1.0.0:', type=('build', 'run'), when='@0.7.0:+docs')
+    depends_on('py-nbsphinx@0.5.0:', type=('build', 'run'), when='@0.7.0:0.8.0+docs')
+    depends_on('py-nbsphinx@0.8.0:', type=('build', 'run'), when='@0.9.0:+docs')
+    depends_on('py-sphinxcontrib-bibtex@1.0.0:', type=('build', 'run'), when='@0.7.0:+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-elephant/package.py]:
```suggestion
    depends_on('py-sphinxcontrib-bibtex@1.0.0', type=('build', 'run'), when='@0.7.0:+docs')
```

File: var/spack/repos/builtin/packages/py-elephant/package.py [hunk 5]
```diff
@@ -3,27 +3,63 @@
...
-    depends_on('py-neo@0.3.4:',         type=('build', 'run'))  # > 0.3.3 ?
-    depends_on('py-numpy@1.8.2:',       type=('build', 'run'))
-    depends_on('py-quantities@0.10.1:', type=('build', 'run'))
-    depends_on('py-scipy@0.14.0:',      type=('build', 'run'))
-    depends_on('py-pandas@0.14.1:',     type=('build', 'run'), when='+pandas')
-    depends_on('py-numpydoc@0.5:',      type=('build', 'run'), when='+doc')
-    depends_on('py-sphinx@1.2.2:',      type=('build', 'run'), when='+doc')
+    variant('docs', default=False, description='Install documentation dependencies')
+    variant('pandas', default=False, description='Build with pandas', when='@0.3.0:0.4.1')
+    variant('extras', default=True, description='Build with extras for GPFA, ASSET', when='@0.6.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-elephant/package.py]: Personally I would make this False by default. That would match the behavior of `pip install elephant` and minimize build times.

Comment (Moritz-Alexander-Kern) [reply var/spack/repos/builtin/packages/py-elephant/package.py]: good point, see [6aeb825](https://github.com/spack/spack/pull/30613/commits/6aeb825282d9608baa37ef5cf68193fadbb04c8d)

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**